rioxarray/rioxarray.py

```diff
@@ -1051,8 +1051,81 @@ def slice_xy(self, minx, miny, maxx, maxy):
         subset.attrs["transform"] = tuple(self.transform(recalc=True))
         return subset
 
-    def clip_box(self, minx, miny, maxx, maxy, auto_expand=False, auto_expand_limit=3):
-        """Clip the :class:`xarray.DataArray` by a bounding box.
+    def pad_xy(self, minx, miny, maxx, maxy, resolution_x, resolution_y):
+        """Pad the array to x,y bounds.
+
+        Parameters
+        ----------
+        minx: float
+            Minimum bound for x coordinate.
+        miny: float
+            Minimum bound for y coordinate.
+        maxx: float
+            Maximum bound for x coordinate.
+        maxy: float
+            Maximum bound for y coordinate.
+        resolution_x: float
+            Resolution of x coordinate.
+        resolution_y: float
+            Resolution of y coordinate.
+
+
+        Returns
+        -------
+        DataArray: A padded :class:`xarray.DataArray` object.
+
+        """
+        left, bottom, right, top = self._internal_bounds()
+        y_before = y_after = 0
+        x_before = x_after = 0
+        y_coord = self._obj[self.y_dim]
+        x_coord = self._obj[self.x_dim]
+
+        if top - resolution_y < maxy:
+            new_y_coord = np.arange(bottom, maxy, -resolution_y)[::-1]
+            y_before = len(new_y_coord) - len(y_coord)
+            y_coord = new_y_coord
+            top = y_coord[0]
+        if bottom + resolution_y > miny:
+            new_y_coord = np.arange(top, miny, resolution_y)
+            y_after = len(new_y_coord) - len(y_coord)
+            y_coord = new_y_coord
+            bottom = y_coord[-1]
+
+        if left - resolution_x > minx:
+            new_x_coord = np.arange(right, minx, -resolution_x)[::-1]
+            x_before = len(new_x_coord) - len(x_coord)
+            x_coord = new_x_coord
+            left = x_coord[0]
+        if right + resolution_x < maxx:
+            new_x_coord = np.arange(left, maxx, resolution_x)
+            x_after = len(new_x_coord) - len(x_coord)
+            x_coord = new_x_coord
+            right = x_coord[-1]
+
+        superset = self._obj.pad(
+            pad_width={
+                self.x_dim: (x_before, x_after),
+                self.y_dim: (y_before, y_after),
+            },
+            constant_values=np.nan,
```

Comment:
I haven't tested this out yet - does this fill the generated area with `NaN`?

Reply:
Yes, you can see it in the `clip_box.ipynb` notebook.